pages/blog/json-schema-beyond-validation.md

```diff
@@ -0,0 +1,151 @@
+authors:
+  - name: Jonas Lagoni
+    photo: /img/avatars/jonaslagoni.webp
+    link: https://github.com/jonaslagoni
+    byline: AsyncAPI Maintainer
+featured: true
+---
+
+What is JSON Schema, why is it important and why is it so hard to use beyond validation? 
+
+For those unfamiliar with Asyncapi [we use a superset of JSON Schema](https://www.asyncapi.com/docs/specifications/2.0.0#schemaObject) as the default format for defining operation payloads, headers, channel parameter schemas, etc.
+
+Even though we allow other formats such as Avro, OpenAPI 3.x and Swagger 2.x, RAML schemas in its place, as soon as it hits the parser (which most tooling utilizes), said formats are converted to JSON Schema draft 7 to ensure a [common structure for tooling](https://github.com/asyncapi/parser-js/blob/826b36922260254ba23d162cda309fc72f552c49/lib/models/message.js#L20). 
+
+However, in tooling, many times you do not want to validate data, but know the structure of the data. For example for the payload of messages, in specific programming languages, how do you represent such data so it is easier to interact with, such as classes that represent the payload? 
+
+## Quick intro to JSON Schema
+
+Let's try and take a look at an example. Given the following, we have a schema representing the validation rules that our data should comply with. 
+
+<figure>
+  <img src="../img/posts/jonaslagoni/json-schema-beyond-validation/json-schema-process.webp" title="Processor setup"/>
+  <figcaption className="text-center text-gray-400 text-sm">Displays the overall process of validating data using JSON Schema.</figcaption>
+</figure>
+
+The JSON Schema defines that the JSON data should be an object, which requires a property called `someRequiredProperty` to always be present and an optional property called `someOptionalProperty`. `someRequiredProperty` should validate against an integer and and `someOptionalProperty` against an arbitrary string. The schema also dictates that no additional properties (`"additionalProperties": false`) may be allowed. There is also some metadata defined, called `$id` and `$schema`, but they are not important for this example.
+
+If we then take a look at the example [data instances](https://datatracker.ietf.org/doc/html/draft-handrews-json-schema-01#section-4.2), the first one contains the required property, and the second one both the required and the optional property.
+
+The data and the JSON Schema can then together, validate whether the data is an instance of the schema i.e. validate whether the data comply with the validation rules and giving a simple true or false statement whether they are compatible.
+
+This is an extremely powerful tool which allows you to create complex validation rules for data, that would not otherwise be easy to do, if you used other specification which defines the data model using [TypeSchema](https://typeschema.org/), [JTD](https://datatracker.ietf.org/doc/html/rfc8927), etc.
```

Comment:
Just to emphasize the sentence and to clarify what we are talking about (leading to zero confusion):

```suggestion
JSON Schema is an extremely powerful tool which allows you to create complex validation rules for data, that would not otherwise be easy to do, if you used other specification which defines the data model using [TypeSchema](https://typeschema.org/), [JTD](https://datatracker.ietf.org/doc/html/rfc8927), etc.
```